lib/oauth-db.js

```diff
@@ -0,0 +1,42 @@
+var Boom = require('boom');
+var crypto = require('crypto');
+var Hoek = require('hoek');
+
+var OAuthDB = function OAuthDB(clients) {
+  this.clients = {};
+  clients.forEach(function(client) {
+    this.clients[client.client_id] = client;
+  }, this);
+  this.auth_codes = {};
+};
+
+OAuthDB.prototype.get_client = function(client_id, callback) {
+  if (!this.clients[client_id]) {
+    return process.nextTick(function() {
+      callback(Boom.badRequest('invalid client_id'));
+    });
+  }
+
+  process.nextTick(function() {
+    callback(null, this.clients[client_id]);
+  }.bind(this));
+};
+
+OAuthDB.prototype.generate_auth_code = function(client_id, user_id, scopes, expires_at, callback) {
+  crypto.randomBytes(32, function(err, random) {
+    Hoek.assert(!err, err);
```

Comment:
Not super familiar with `hoek.assert` yet but this will throw an exception, yes? If so, I'm curious why we would throw here rather than passing it to the callback ala

```js
if (err) return callback(err);
```